src/wallet/coin_selection.rs

```diff
@@ -534,6 +534,49 @@ impl BranchAndBoundCoinSelection {
     }
 }
 
+struct WasteMetric;
+
+impl WasteMetric {
+    fn calculate_waste(
+        selected: Vec<WeightedUtxo>,
+        cost_of_change: Option<u64>,
+        amount_needed: u64,
+        fee_rate: FeeRate,
+        long_term_fee_rate: FeeRate,
+    ) -> i64 {
```

Comment:
Instead of returning `i64` why not return just return `Self`? And have `i64` or whatever as member function of `WasteMetric`. If we are defining a struct anyway, lets use it. :) 

In this way we can also use this inside `CoinSelectionResult` later. This seems to me like a very nice tool to compare different selection algorithm and we can expose a simple API to users with some default configs (like long term fee rate etc).